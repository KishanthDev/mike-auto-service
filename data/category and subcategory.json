[
  {
    "category": "Automotive",
    "subcategories": ["Car Repair", "Car Sales", "Tires", "Auto Parts"]
  },
  {
    "category": "Health & Medical",
    "subcategories": [
      "Hospitals",
      "Clinics",
      "Dentists",
      "Pharmacies",
      "Physiotherapy Centres",
      "Veterinary Clinics"
    ]
  },
  {
    "category": "Education",
    "subcategories": [
      "Schools (primary, secondary)",
      "Colleges & Universities",
      "Tutoring Services",
      "Educational Consultants"
    ]
  },
  {
    "category": "Restaurants & Food",
    "subcategories": [
      "Fast Food",
      "Fine Dining",
      "Cafes",
      "Bakeries",
      "Food Delivery Services"
    ]
  },
  {
    "category": "Shopping & Retail",
    "subcategories": [
      "Clothing Stores",
      "Electronics Stores",
      "Jewellery Showrooms",
      "Grocery Stores",
      "Department Stores",
      "Furniture Showrooms",
      "Pest Control"
    ]
  },
  {
    "category": "Professional Business Services",
    "subcategories": [
      "Chartered Accountants",
      "Lawyers",
      "Marketing Agencies",
      "Business Consultants"
    ]
  },
  {
    "category": "Home Services",
    "subcategories": [
      "Interior Designers",
      "Carpenters",
      "Plumbers",
      "Electricians",
      "Cleaners",
      "Landscapers",
      "Home Renovation"
    ]
  },
  {
    "category": "Travel & Transportation",
    "subcategories": [
      "Airlines",
      "Hotels",
      "Car Rentals",
      "Travel Agencies",
      "Public Transport"
    ]
  },
  {
    "category": "Real Estate",
    "subcategories": [
      "Builders and Developers",
      "Civil Contractors",
      "Real Estate Agents",
      "Property Management"
    ]
  },
  {
    "category": "Financial Services",
    "subcategories": [
      "Banks",
      "Loan Agencies",
      "Insurance Agencies",
      "Investment Services",
      "Credit Unions"
    ]
  },
  {
    "category": "Government & Community",
    "subcategories": [
      "Local Government Offices",
      "Community Centers",
      "Libraries",
      "Post Offices"
    ]
  },
  {
    "category": "Entertainment & Media",
    "subcategories": [
      "Movie Theaters",
      "Music Stores",
      "Event Management",
      "Convention Halls",
      "Media Production"
    ]
  },
  {
    "category": "Sports & Recreation",
    "subcategories": [
      "Gyms",
      "Sports Leagues",
      "Parks",
      "Recreation Centers",
      "Sports Equipment Stores"
    ]
  },
  {
    "category": "Pets & Animals",
    "subcategories": [
      "Pet Stores",
      "Veterinarians",
      "Pet Grooming",
      "Animal Shelters",
      "Pet Training"
    ]
  },
  {
    "category": "Computers & Mobile",
    "subcategories": [
      "Internet Service Providers",
      "Computer Sales and Repair",
      "Mobile Sales and Repair",
      "Web Design",
      "Software Solutions"
    ]
  },
  {
    "category": "Safety & Security",
    "subcategories": [
      "Private Security Services",
      "Fire Departments",
      "Police Stations",
      "Safety Equipment"
    ]
  }
]
