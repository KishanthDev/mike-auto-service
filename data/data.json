{
    "businessHours": [
      { "day": "Monday - Friday", "hours": "7:30 AM - 6:00 PM" },
      { "day": "Saturday", "hours": "8:00 AM - 4:00 PM" },
      { "day": "Sunday", "hours": "Closed" }
    ],
    "certifications": ["AUDI", "BMW", "Honda", "Toyota", "Ford"],
    "images": [
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952",
      "https://via.placeholder.com/150/92c952"
    ],
    "reviews": [
      {
        "author": "Sarah J.",
        "rating": 5,
        "content": "Mike's team fixed my Honda's transmission issue quickly and for much less than the dealership quoted. Very honest and transparent about costs."
      },
      {
        "author": "David T.",
        "rating": 5,
        "content": "I've been coming here for 5 years and have always had excellent service. They explain everything clearly and never push unnecessary repairs."
      }
    ],
    "services": [
      {
        "title": "Brake Service",
        "description": "Complete brake system inspection, repair and replacement. Lifetime warranty on most brake pads.",
        "price": "Starting at $89.99"
      },
      {
        "title": "Engine Repair",
        "description": "Diagnostic and repair services for all engine types. Factory-trained technicians.",
        "price": "Diagnosis: $89.99"
      },
      {
        "title": "Transmission Service",
        "description": "Transmission fluid change, repair and rebuilding services with 24-month warranty.",
        "price": "Fluid change: $149.99"
      },
      {
        "title": "Electrical Systems",
        "description": "Computer diagnostics, battery service, alternator replacement and more.",
        "price": "Starting at $69.99"
      }
    ]
  }
  