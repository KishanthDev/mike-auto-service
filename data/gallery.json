[
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952",
    "https://via.placeholder.com/150/92c952"
  ]
  