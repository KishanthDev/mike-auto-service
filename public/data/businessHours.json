[
    { "day": "Monday - Friday", "hours": "7:30 AM - 6:00 PM" },
    { "day": "Saturday", "hours": "8:00 AM - 4:00 PM" },
    { "day": "Sunday", "hours": "Closed" }
  ]
  