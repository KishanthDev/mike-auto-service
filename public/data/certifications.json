["AUDI", "BMW", "Honda", "Toyota", "Ford"]
