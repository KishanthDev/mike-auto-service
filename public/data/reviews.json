[
    {
      "author": "Sarah J.",
      "rating": 5,
      "content": "Mike's team fixed my Honda's transmission issue quickly and for much less than the dealership quoted. Very honest and transparent about costs."
    },
    {
      "author": "David T.",
      "rating": 5,
      "content": "I've been coming here for 5 years and have always had excellent service. They explain everything clearly and never push unnecessary repairs."
    }
  ]
  