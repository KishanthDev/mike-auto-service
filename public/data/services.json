[
    {
      "title": "Brake Service",
      "description": "Complete brake system inspection, repair and replacement. Lifetime warranty on most brake pads.",
      "price": "Starting at $89.99"
    },
    {
      "title": "Engine Repair",
      "description": "Diagnostic and repair services for all engine types. Factory-trained technicians.",
      "price": "Diagnosis: $89.99"
    },
    {
      "title": "Transmission Service",
      "description": "Transmission fluid change, repair and rebuilding services with 24-month warranty.",
      "price": "Fluid change: $149.99"
    },
    {
      "title": "Electrical Systems",
      "description": "Computer diagnostics, battery service, alternator replacement and more.",
      "price": "Starting at $69.99"
    }
  ]
  