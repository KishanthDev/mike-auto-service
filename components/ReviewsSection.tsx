"use client";

import React from "react";
import { Card, CardBody } from "@heroui/card";
import reviews from "../data/reviews.json";

export default function ReviewsSection() {
  return (
    <section>
      <h2 className="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white">
        Customer Reviews
      </h2>

      <div className="space-y-4 mt-4 md:mt-6">
        {reviews.map((review, idx) => (
          <Card
            key={idx}
            className="bg-white dark:bg-gray-800 border border-gray-200 dark:border-gray-700"
          >
            <CardBody className="p-4 md:p-6 text-sm md:text-base text-gray-700 dark:text-gray-300">
              <div className="flex flex-col sm:flex-row sm:justify-between gap-2 mb-2">
                <span className="font-semibold text-gray-900 dark:text-white">
                  {review.author}
                </span>
                <div className="text-yellow-400">
                  {"â˜…".repeat(review.rating)}
                </div>
              </div>
              <p>{review.content}</p>
            </CardBody>
          </Card>
        ))}
      </div>
    </section>
  );
}
