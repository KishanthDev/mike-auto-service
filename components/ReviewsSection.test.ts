import React from "react";
import { render, screen } from "@testing-library/react";
import ReviewsSection from "./ReviewsSection";
import { Review } from "@/types/data";

// Mock the Card components using React.createElement
jest.mock("@heroui/card", () => ({
  Card: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) =>
    React.createElement(
      "div",
      {
        className: `mock-card ${className}`,
        "data-testid": "card",
      },
      children,
    ),
  CardBody: ({
    children,
    className,
  }: {
    children: React.ReactNode;
    className?: string;
  }) =>
    React.createElement(
      "div",
      {
        className: `mock-card-body ${className}`,
        "data-testid": "card-body",
      },
      children,
    ),
}));

describe("ReviewsSection Component", () => {
  const mockReviews: Review[] = [
    {
      author: "John Doe",
      rating: 5,
      content: "Excellent service!",
    },
    {
      author: "Jane Smith",
      rating: 4,
      content: "Very professional team",
    },
  ];

  beforeEach(() => {
    render(React.createElement(ReviewsSection, { reviews: mockReviews }));
  });

  it("displays the correct heading", () => {
    const heading = screen.getByRole("heading", { level: 2 });
    expect(heading).toBeInTheDocument();
    expect(heading).toHaveTextContent("Customer Reviews");
    expect(heading).toHaveClass("text-2xl");
    expect(heading).toHaveClass("border-b-2");
    expect(heading).toHaveClass("border-red-600");
  });

  it("renders all review cards", () => {
    const cards = screen.getAllByTestId("card");
    expect(cards).toHaveLength(mockReviews.length);
  });

  it("applies correct card styling", () => {
    const cards = screen.getAllByTestId("card");
    cards.forEach((card) => {
      expect(card).toHaveClass("bg-white");
      expect(card).toHaveClass("dark:bg-gray-800");
      expect(card).toHaveClass("border");
    });
  });

  it("displays correct review content", () => {
    mockReviews.forEach((review) => {
      expect(screen.getByText(review.author)).toBeInTheDocument();
      expect(screen.getByText(review.content)).toBeInTheDocument();
    });
  });

  it("displays correct star ratings", () => {
    const fiveStars = screen.getAllByText("★★★★★");
    const fourStars = screen.getAllByText("★★★★");

    expect(fiveStars.length).toBe(1);
    expect(fourStars.length).toBe(1);
    expect(fiveStars[0]).toHaveClass("text-yellow-400");
    expect(fourStars[0]).toHaveClass("text-yellow-400");
  });

  it("applies correct responsive classes", () => {
    const cardBodies = screen.getAllByTestId("card-body");
    cardBodies.forEach((body) => {
      expect(body).toHaveClass("p-4");
      expect(body).toHaveClass("md:p-6");
      expect(body).toHaveClass("text-sm");
      expect(body).toHaveClass("md:text-base");
    });
  });

  it("matches snapshot", () => {
    const { asFragment } = render(
      React.createElement(ReviewsSection, { reviews: mockReviews }),
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
