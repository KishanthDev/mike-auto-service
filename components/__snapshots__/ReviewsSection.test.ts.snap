// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsSection Component matches snapshot 1`] = `
<DocumentFragment>
  <section>
    <h2
      class="text-2xl md:text-3xl font-bold border-b-2 border-red-600 pb-2 text-[#1a3c6e] dark:text-white"
    >
      Customer Reviews
    </h2>
    <div
      class="space-y-4 mt-4 md:mt-6"
    >
      <div
        class="mock-card bg-white dark:bg-gray-800 border border-gray-200 dark:border-gray-700"
        data-testid="card"
      >
        <div
          class="mock-card-body p-4 md:p-6 text-sm md:text-base text-gray-700 dark:text-gray-300"
          data-testid="card-body"
        >
          <div
            class="flex flex-col sm:flex-row sm:justify-between gap-2 mb-2"
          >
            <span
              class="font-semibold text-gray-900 dark:text-white"
            >
              John Doe
            </span>
            <div
              class="text-yellow-400"
            >
              ★★★★★
            </div>
          </div>
          <p>
            Excellent service!
          </p>
        </div>
      </div>
      <div
        class="mock-card bg-white dark:bg-gray-800 border border-gray-200 dark:border-gray-700"
        data-testid="card"
      >
        <div
          class="mock-card-body p-4 md:p-6 text-sm md:text-base text-gray-700 dark:text-gray-300"
          data-testid="card-body"
        >
          <div
            class="flex flex-col sm:flex-row sm:justify-between gap-2 mb-2"
          >
            <span
              class="font-semibold text-gray-900 dark:text-white"
            >
              Jane Smith
            </span>
            <div
              class="text-yellow-400"
            >
              ★★★★
            </div>
          </div>
          <p>
            Very professional team
          </p>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
